### PR TITLE
Fix/protect emit calls

### PR DESCRIPTION
**Pull request by roggervalf**

### CODE REVIEW

File: src/test/test_events.ts [hunk 1]
```diff
@@ -89,6 +89,29 @@ describe('events', function() {
     await worker.close();
   });
 
+  it('emits error event when there is an error on other events', async function() {
+    const worker = new Worker(queueName, async job => {}, {
+      drainDelay: 1,
+    });
+
+    worker.once('completed', (job: any) => {
+      console.log(job.bar.id);
```
Comment (manast) [inline src/test/test_events.ts:98]: We should not console.log in test cases.

File: src/test/test_events.ts [hunk 2]
```diff
@@ -89,6 +89,29 @@ describe('events', function() {
...
+    });
+
+    const error = new Promise<void>(resolve => {
+      worker.once('error', resolve);
+    });
+
+    await queue.add('test', { foo: 'bar' });
+
+    await error;
```
Comment (manast) [inline src/test/test_events.ts:107]: What is causing the error event? I am re-reading the test several times but cannot find the part that triggers the error 😂

Comment (manast) [reply src/test/test_events.ts:107]: ah ok, it is trying to access id on bar on the job instance (since bar is undefined),

Comment (roggervalf) [reply src/test/test_events.ts:107]: yes, jejeje 😆

**Pull request merged**